cli/kubectl-hyperkube

```diff
@@ -330,18 +347,33 @@ def get_cluster_metadata(ctx, cluster):
     """Get cluster metadata for a give cluster name"""
 
     try:
-        r = requests.get(
-            (f'{ctx.obj["url"]}/{ctx.obj["stage"]}'
-             f'/clusters/get-cluster-metadata?cluster_name={cluster}'),
-            headers=ctx.obj['headers']
-        )
+        r = _build_request(ctx, "GET", "/clusters/get-cluster-metadata",
+                           query_string=f'?cluster_name={cluster}')
         hkc_print(r.json())
         if r.status_code == 404:
             sys.exit(1)
     except requests.exceptions.RequestException as err:
         print(f'Request error: {err}')
 
 
+def _build_request(ctx, verb, api_endpoint,
+                   query_string="", headers={}, data=None):
+    """Build http request"""
+
+    base_url = ctx.obj['base_url']
+    url = base_url + api_endpoint + query_string
+    auth = ctx.obj['auth']
+
+    # Ability to add headers beyond api key
+    merged_headers = {**ctx.obj['headers'], **headers}
```

Comment:
why not `headers.update(**ctx.obj['headers'])` ?

Reply:
That will keep existing keys as is?

Reply:
Nice! seems to have same behavior and easier to read! ty!